neuraxle/hyperparams/space.py

```diff
@@ -282,31 +237,45 @@ class HyperparameterSpace(RecursiveDict):
     def __init__(self, *args, separator=None, **kwds):
         super().__init__(*args, separator=separator, **kwds)
 
+    def _patch_arg(self, arg):
+        """
+        Override of the RecursiveDict's default _patch_arg(arg) method.
+        """
+        if hasattr(arg, 'dist') and isinstance(arg.dist, rv_generic):
+            if isinstance(arg.dist, rv_discrete):
+                return ScipyDiscreteDistributionWrapper(arg), True
+            if isinstance(arg.dist, rv_continuous):
+                return ScipyContinuousDistributionWrapper(arg), True
+        else:
+            return arg, False
+
     def rvs(self) -> 'HyperparameterSamples':
         """
         Sample the space of random variables.
 
         :return: a random HyperparameterSamples, sampled from a point of the present HyperparameterSpace.
         """
         new_items = []
-        for k, v in self.items():
-            if isinstance(v, HyperparameterDistribution) or isinstance(v, HyperparameterSpace):
+        for k, v in self.iter_flat():
+            if isinstance(v, HyperparameterDistribution):
                 v = v.rvs()
             new_items.append((k, v))
         return HyperparameterSamples(new_items)
 
+    # TODO : The following functions aren't used, or tested, anywhere. They should work though.
+
     def nullify(self):
         new_items = []
-        for k, v in self.items():
+        for k, v in self.iter_flat():
             if isinstance(v, HyperparameterDistribution) or isinstance(v, HyperparameterSpace):
                 v = v.nullify()
             new_items.append((k, v))
         return HyperparameterSamples(new_items)
 
     def narrow_space_from_best_guess(
-        self,
-        best_guesses: 'HyperparameterSpace',
-        kept_space_ratio: float = 0.5
+            self,
+            best_guesses: 'HyperparameterSpace',
+            kept_space_ratio: float = 0.5
```

Comment:
Note that this function may be deleted in the future. It is probably never used except in tests. Documentation notebooks may need to be updated accordingly as well. Possibly resolve to issue.

Reply:
#447